bundle/src/bundle_meta.rs

```diff
@@ -0,0 +1,102 @@
+//! BundleMeta is the contents of meta.json
+//! - We create a BundleMeta (current CLI version) on upload
+//! - We read BundleMetaV_* (old versions), incrementally, during parsing on services side
+//!
+
+use codeowners::CodeOwners;
+use context::repo::BundleRepo;
+use serde::{Deserialize, Serialize};
+
+use crate::{files::FileSet, CustomTag, MapType, Test};
+
+#[cfg(feature = "wasm")]
+use wasm_bindgen::prelude::*;
+
+pub const META_VERSION: &str = "1";
+
+// 0.5.29 was first version to include bundle_upload_id and serves as the base
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[cfg_attr(feature = "wasm", wasm_bindgen(getter_with_clone))]
+pub struct BundleMetaBaseProps {
+    pub version: String,
+    pub cli_version: String,
+    pub org: String,
+    pub repo: BundleRepo,
+    pub bundle_upload_id: String,
+    pub tags: Vec<CustomTag>,
+    pub file_sets: Vec<FileSet>,
+    pub envs: MapType,
+    pub upload_time_epoch: u64,
+    pub test_command: Option<String>,
+    pub os_info: Option<String>,
+    pub quarantined_tests: Vec<Test>,
+    pub codeowners: Option<CodeOwners>,
+}
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[cfg_attr(feature = "wasm", wasm_bindgen(getter_with_clone))]
+pub struct BundleMetaV0_5_29 {
+    #[serde(flatten)]
+    pub base_props: BundleMetaBaseProps,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[cfg_attr(feature = "wasm", wasm_bindgen(getter_with_clone))]
+pub struct BundleMetaJunitProps {
+    pub num_files: usize,
+    pub num_tests: usize,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[cfg_attr(feature = "wasm", wasm_bindgen(getter_with_clone))]
+pub struct BundleMeta {
+    #[serde(flatten)]
+    pub base_props: BundleMetaBaseProps,
+    #[serde(flatten)]
+    pub junit_props: Option<BundleMetaJunitProps>,
+}
+
+//// Add new versions here and rename the above struct ////
+
+pub type BundleMetaV0_5_34 = BundleMeta;
```

Comment:
Soliciting feedback on some of the versioning conventions here. We need to be backwards compatible as much as possible for these structs, since we'll be parsing from older client versions. Right now, I just made any of the fields we added since `v0.5.29` (first inclusion of `bundle_upload_id`) optional and added comments. If desired, I can work back to older versions as well.

I can also look into adding snapshot tests or similar in jest to make sure we handle older sample client output properly.